src/interface/builder.py

```diff
@@ -894,89 +927,61 @@ def setMatrix(self):
-            return firstpassagetimes
-        
-        if self.solveToggle == 8:
-            firstpassagetimes, info = cg(self.rate_matrix_csr, self.b, M=self.rate_matrix_inverse)
-            return firstpassagetimes
-        
-        if self.solveToggle == 9:
-            firstpassagetimes, info = lgmres(self.rate_matrix_csr, self.b)
-            return firstpassagetimes
-        
-        if self.solveToggle == 10:
-            firstpassagetimes, info = lgmres(self.rate_matrix_csr, self.b, M=self.rate_matrix_inverse)
-            return firstpassagetimes
-
-        firstpassagetimes = spsolve(self.rate_matrix_csr, self.b)
+            firstpassagetimes, info = bicg(self.rate_matrix_csr, self.b, x0 = x0)
+        elif self.solveToggle == 2:
+            firstpassagetimes, info = bicg(self.rate_matrix_csr, self.b, M=self.rate_matrix_inverse,  x0 = x0)
+        elif self.solveToggle == 3:
+            firstpassagetimes, info = bicgstab(self.rate_matrix_csr, self.b ,  x0 = x0)
+        elif self.solveToggle == 4:
+            firstpassagetimes, info = bicgstab(self.rate_matrix_csr, self.b, M=self.rate_matrix_inverse,  x0 = x0)
+        elif self.solveToggle == 5:
+            firstpassagetimes, info = gmres(self.rate_matrix_csr, self.b,  x0 = x0)
+        elif self.solveToggle == 6:
+            firstpassagetimes, info = gmres(self.rate_matrix_csr, self.b, M=self.rate_matrix_inverse,  x0 = x0)
+        elif self.solveToggle == 7:
+            firstpassagetimes, info = cg(self.rate_matrix_csr, self.b,  x0 = x0)
+        elif self.solveToggle == 8:
+            firstpassagetimes, info = cg(self.rate_matrix_csr, self.b, M=self.rate_matrix_inverse,  x0 = x0)
+        elif self.solveToggle == 9:
+            firstpassagetimes, info = lgmres(self.rate_matrix_csr, self.b,  x0 = x0)
+        elif self.solveToggle == 10:
+            firstpassagetimes, info = lgmres(self.rate_matrix_csr, self.b, M=self.rate_matrix_inverse,  x0 = x0)
+        else:
+            firstpassagetimes = spsolve(self.rate_matrix_csr, self.b)
+        self.matrixTime = time.time() - startTime
         return firstpassagetimes
-        
-    def averageTimeFromInitial(self, bimolecular=False, printMeanTime=True):
 
-        startTime = time.time()
+    def averageTimeFromInitial(self, bimolecular=False, printMeanTime=True):
 
         if self.build.verbosity:
             print "Starting to solve the matrix equation.."
-            
-        firstpassagetimes = self.averageTime()  # spsolve(self.rate_matrix_csr  , self.b)        
+        firstpassagetimes = self.averageTime()  # spsolve(self.rate_matrix_csr  , self.b)
+        mfpt = self.averageTimeFromInitial_givenfirstpassagetimes(firstpassagetimes =firstpassagetimes, bimolecular =bimolecular, printMeanTime =printMeanTime)
+        return mfpt
 
+    def averageTimeFromInitial_givenfirstpassagetimes(self, firstpassagetimes, bimolecular = False, printMeanTime=  True) :
```

Comment:
rename as "weightedPassageTime" 

Move the print statement to averageTimeFromIntial, printing it here is awkward because the solving already happened. So the print argument will disappear from this function.